src/NodaTime.Test/Annotations/ReflectionTest.cs

```diff
@@ -26,9 +27,17 @@ public void ConversionNamesMatchTargetType()
                 .SelectMany(m => m.DeclaredMethods)
                 .Where(m => m.IsPublic)
                 .Where(m => conversionName.IsMatch(m.Name))
-                .Where(m => !m.Name.StartsWith($"To{m.ReturnType.Name}"));
+                .Where(NameIsInvalid);
 
             TestHelper.AssertNoFailures(badMethods, m => $"{m.DeclaringType?.Name}.{m.Name}", TestExemptionCategory.ConversionName);
+
+            bool NameIsInvalid(MethodInfo method)
```

Comment:
Maybe `IsUnexpected` or `IsInconsistentWithReturnType`? If this test fails, the method name is definitely valid, it's just not what we wanted :)

Reply:
The latter is nice. Done.